src/Lemon/Containers/Converters/Ncch2Binary.cs

```diff
@@ -0,0 +1,237 @@
+
+            var binary = (stream != null) ? new BinaryFormat(stream) : new BinaryFormat();
+            var writer = new DataWriter(binary.Stream);
+
+            Node root = source.Root;
+
+            // Ncch header data has a length of 0x200 before binary data.
+            writer.Write(new byte[0x200]);
+            writer.Stream.Position = 0;
+
+            // First we write the header signature
+            writer.Write(source.Header.Signature);
+
+            byte[] magicId = new byte[] { 0x4E, 0x43, 0x43, 0x48 };
+            writer.Write(magicId);
+
+            // We'll write the total size at last.
+            writer.Stream.Position += 0x4;
+
+            writer.Write(source.Header.PartitionId);
+            writer.Write(source.Header.MakerCode);
+            writer.Write(source.Header.Version);
+
+            if (source.Header.Flags[7] == 0x20) {
+                throw new FormatException("Value 0x20 for Flag 0x07 not supported yet");
+            }
+            else {
+                writer.Stream.Position += 0x4;
+            }
+
+            writer.Write(source.Header.ProgramId);
+
+            writer.Stream.Position += 0x10; // Reserved
+
+            WriteSHA256(writer, root.Children["logo.bin"]);
+            writer.Write(source.Header.ProductCode.PadRight(0x10, '\0'));
+            WriteSHA256(writer, root.Children["extended_header"]);
+            writer.Write((int)(root.Children["extended_header"]?.Stream?.Length ?? 0));
+
+            writer.Stream.Position += 0x4; // Reserved
+
+            for (int i = 0; i < source.Header.Flags.Length; i++) {
+                byte flag = source.Header.Flags[i];
+                writer.Write(flag);
+            }
+
+            // Now we begin writing the NCCH content
+            writer.Stream.Position = 0x200;
+            WriteFile(writer, root.Children["extended_header"], false);
+            WriteFile(writer, root.Children["access_descriptor"], false);
+
+            writer.Stream.Position = 0x190;
+            WriteOffsetSizeAndData(writer, root.Children["sdk_info.txt"], source.Header.SdkInfoOffset);
+            WriteOffsetSizeAndData(writer, root.Children["logo.bin"], source.Header.LogoOffset);
+            WriteOffsetSizeAndData(writer, root.Children["system"], source.Header.SystemOffset, source.Header.SystemHashSize, true);
+
+            writer.Stream.Position += 0x4; // Reserved
+            WriteOffsetSizeAndData(writer, root.Children["rom"], source.Header.RomOffset, source.Header.RomHashSize, true);
+
+            writer.Stream.Position += 0x4; // Reserved
+
+            WriteSHA256(writer, root.Children["system"], source.Header.SystemHashSize, true);
+            WriteSHA256(writer, root.Children["rom"], source.Header.RomHashSize, true);
+
+            // Write the full size of the binary (in units) in 0x104
+            writer.Stream.Position = 0x104;
+            writer.Write((int)writer.Stream.Length / NcchHeader.Unit);
```

Comment:
I think that if the division is not exact, then we will need to add padding and write a higher number. For instance, if Stream is 768 bytes, we need to write `2` but right now it would be `768 / 512 = 1`.
We can get this done by simply calling before this line: `writer.WritePadding(0, NcchHeader.Unit);`

It will do nothing if the length is already a multiple of 512 (Unit), otherwise it will write padding and then we will get the correct stream length.